### ISSUE TITLE
"Not implemented" crash when parsing SqlProvider solution

### ISSUE THREAD

**Issue by knocte** [fsprojects/FSharpLint#670]
### Repro steps

Please provide the steps required to reproduce the problem

1. Clone https://github.com/fsprojects/SqlProvider
2. Build it invoking build.sh.
3. Run fsharplint on its sln file.

### Expected behavior

Should not give any errors.

### Actual behavior

One of the projects throws an error:
```
========== Linting /Users/knocte/Documents/Code/SQLProvider/paket-files/sourcefiles/fsprojects/FSharp.TypeProviders.SDK/src/ProvidedTypes.fs ==========
Failed to parse file /Users/knocte/Documents/Code/SQLProvider/paket-files/sourcefiles/fsprojects/FSharp.TypeProviders.SDK/src/ProvidedTypes.fs
Exception Message:
Not implemented
Exception Stack Trace:
   at FSharpLint.Rules.Helper.Naming.getPatternIdents[T](AccessControlLevel accessibility, FSharpFunc`2 getIdents, Boolean argsAreParameters, SynPat pattern) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Rules/Conventions/Naming/NamingHelper.fs:line 365
   at FSharpLint.Rules.ParameterNames.getIdentifiers(AstNodeRuleParams args) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Rules/Conventions/Naming/ParameterNames.fs:line 44
   at FSharpLint.Rules.ParameterNames.Pipe #1 input at line 55@57.Invoke(AstNodeRuleParams args)
   at FSharpLint.Rules.Helper.Naming.astNodeRunner@162.Invoke(AstNodeRuleParams args) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Rules/Conventions/Naming/NamingHelper.fs:line 162
   at FSharpLint.Framework.Rules.runAstNodeRule(RuleMetadata`1 rule, AstNodeRuleParams config) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Framework/Rules.fs:line 95
   at FSharpLint.Application.Lint.astNodeSuggestions@147-1.Invoke(RuleMetadata`1 rule) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Application/Lint.fs:line 147
   at Microsoft.FSharp.Collections.ArrayModule.Collect[T,TResult](FSharpFunc`2 mapping, T[] array) in D:\a\_work\1\s\src\FSharp.Core\array.fs:line 147
   at FSharpLint.Application.Lint.astNodeSuggestions@129.Invoke(Tuple`2 tupledArg) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Application/Lint.fs:line 147
   at Microsoft.FSharp.Collections.ArrayModule.Collect[T,TResult](FSharpFunc`2 mapping, T[] array) in D:\a\_work\1\s\src\FSharp.Core\array.fs:line 147
   at FSharpLint.Application.Lint.runAstNodeRules(RuleMetadata`1[] rules, GlobalRuleConfig globalConfig, FSharpOption`1 typeCheckResults, String filePath, String fileContent, String[] lines, Node[] syntaxArray) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Application/Lint.fs:line 129
   at FSharpLint.Application.Lint.lint(LintInfo lintInfo, FileParseInfo fileInfo) in /Users/knocte/Documents/Code/FSharpLintLEGACY/src/FSharpLint.Core/Application/Lint.fs:line 226

========== Finished: 0 warnings ==========
```

### Related information

* Operating system: macOS
* Branch: master (nuget pkg: 0.23.7--date20240108-0751.git-6636fb2 )
* .NET Runtime, CoreCLR or Mono Version: 6.0.417
